huntsman/observatory.py

```diff
@@ -316,6 +314,105 @@ def take_flat_fields(self, camera_names=None, alt=None, az=None,
 
         self.logger.info('Finished flat-fielding.')
 
+    def take_dark_fields(self,
+                         exptimes,
+                         wait_interval=1,
+                         camera_names=None,
+                         n_darks=10,
+                         *args, **kwargs
+                         ):
+        """Take dark fields for each camera with all the exposure times
+           in the targets list.
+
+        Args:
+            exptimes_list (list): List of exposure times for darks
+            wait_interval (float, optional): Time in between dark exposures
+            camera_list (list, optional): List of cameras to use for darks
+            n_darks (int, optional): Number of darks to be taken per exptime
+            *args (TYPE): Description
+            **kwargs (TYPE): Description
+
+        """
+
+        if camera_names is None:
+            cameras_all = self.cameras
+        else:
+            cameras_all = {c: self.cameras[c] for c in camera_names}
+
+        self.logger.debug('Using cameras {}'.format(cameras_all))
+
+        image_dir = self.config['directories']['images']
+
+        self.logger.debug("Going to take {} dark fields for each of these exposure times {}".
+                          format(n_darks, exptimes))
+
+        # List to check that the final number of darks is equal to the number
+        # of cameras times the number of exptimes times n_darks.
+        total_exposures = []
+
+        # Loop over cameras.
+        for exptime in exptimes:
+
+            self.status()  # Seems to help with reading coords
+
+            start_time = utils.current_time()
+
+            if isinstance(exptime, u.Quantity):
+                exptime = exptime.to_value(u.second)
+
+            # Loop over exposure times for each camera.
+            for num in range(n_darks):
+
+                self.logger.debug('Darks sequence #{} of exposure time {}s'.
+                                  format(num, exptime))
+
+                camera_events = dict()
+
+                # Take a given number of exposures for each exposure time.
+                for camera in cameras_all.values():
+
+                    self.logger.debug('Camera {} is exposing for {}s'.
+                                      format(camera.uid, exptime))
```

Comment:
This statement isn't quite true yet as we haven't yet started exposing.  Could also change to f-string.